lib/datasource/index.ts

```diff
@@ -80,10 +81,20 @@ function getRawReleases(
 export async function getPkgReleases(
   config: PkgReleaseConfig
 ): Promise<ReleaseResult | null> {
-  const res = await getRawReleases({
-    ...config,
-    lookupName: config.lookupName || config.depName,
-  });
+  const { datasource } = config;
+  const lookupName = config.lookupName || config.depName;
+  let res;
+  try {
+    res = await getRawReleases({
+      ...config,
+      lookupName,
+    });
+  } catch (e) /* istanbul ignore next */ {
+    if (e instanceof DatasourceError) {
+      logger.warn({ datasource, lookupName, err: e.err }, 'Datasource failure');
```

Comment:
🤔 Ok, datasource name is known here. So we can set name here instead of passing to constructor

```suggestion
      e.name = datasource;
      logger.warn({ datasource, lookupName, err: e.err }, 'Datasource failure');
```